### PR TITLE
Docs - add `Customizing Queries` page

### PR DESCRIPTION
**Pull request by Shrugsy**
Related to #964 

- Adds a 'Customizing Queries' page showcasing how to use different `baseQuery` functions for `createApi`, as well as `queryFn` on individual endpoints.
- Relocates existing examples of custom `baseQuery` to this page.

### CODE REVIEW

**Comment by phryneas** [comment]
This looks pretty good. We should probably also add `Meta` in here.

Essentially, `Meta` is information that is not stored in `data` by default, but can be returned from the `baseQuery` to be accessed in `transformResponse`.

It *was* also available in `onSuccess` and `onError`, but that is not the case any more.

We *could* think of re-adding that in the `onQuery` lifecycle, but that would require us to include it in the thunk actions and I kinda don't want to add it to the `payload`.

That would be a very fine opportunity to do #844 - we now even have our own use case for that!

**Comment by Shrugsy** [comment]
> This looks pretty good. We should probably also add Meta in here.

Definitely, I completely glossed over that. I've added a Meta example, and also updated the return types for `fetchBaseQuery` to include it's `meta` property.

The `transformResponse` usage I've added for the Meta example is definitely very contrived, but I wanted to show using `meta` in some way

File: docs/usage/rtk-query/customizing-queries.mdx [hunk 1]
```diff
@@ -406,3 +405,96 @@ const api = createApi({
...
+// highlight-start
+const metaBaseQuery: BaseQueryFn<
+  string | FetchArgs,
+  unknown,
+  FetchBaseQueryError,
+  {},
+  Meta & FetchBaseQueryMeta
+> = async (args, api, extraOptions) => {
+  const requestId = uuid()
+  const timestamp = Date.now()
```
Comment (phryneas) [inline docs/usage/rtk-query/customizing-queries.mdx:453]: requestId and timestamp are a bit unfortunate examples as both of those are already existing - the requestId should even be available here as `api.requestId`.
I can't really think of a better example though :sweat_smile: 
So I guess let's leave it like this now and see once a better example comes up in chat or in an issue :laughing:

Comment (Shrugsy) [reply docs/usage/rtk-query/customizing-queries.mdx:453]: Yep, I'm definitely not happy with that example but it's all I could think of :sweat_smile: 

It should definitely be replaced when someone comes up with a better usage for custom `meta`

Comment (phryneas) [reply docs/usage/rtk-query/customizing-queries.mdx:453]: Hmm. In the context of the "retry" wrapper.. maybe the amount of retries?

Comment (Shrugsy) [reply docs/usage/rtk-query/customizing-queries.mdx:453]: Hmm that's a good idea. I'll get some more of your thoughts on it after I have a think on it to try to come up with a feasible example snippet

**Pull request merged**